packages/features/users/components/InvalidAppCredentialsBanner.tsx

```diff
@@ -16,22 +16,37 @@ export function InvalidAppCredentialBanners({ data }: InvalidAppCredentialBanner
   return (
     <div>
       {data.map((app) => (
-        <InvalidAppCredentialBanner key={app.slug} name={app.name} slug={app.slug} />
+        <InvalidAppCredentialBanner id={app.id} key={app.slug} name={app.name} slug={app.slug} />
       ))}
     </div>
   );
 }
 
 export type InvalidAppCredentialBannerProps = {
+  id: number;
   name: string;
   slug: string;
 };
 
-export function InvalidAppCredentialBanner({ name, slug }: InvalidAppCredentialBannerProps) {
+export function InvalidAppCredentialBanner({ name, slug, id }: InvalidAppCredentialBannerProps) {
   const { t } = useLocale();
   const router = useRouter();
+  const utils = trpc.useUtils();
+  const mutation = trpc.viewer.deleteCredential.useMutation({
+    onSuccess: () => {
+      showToast(t("app_removed_successfully"), "success");
+    },
+    onError: () => {
+      showToast(t("error_removing_app"), "error");
+    },
+    async onSettled() {
+      await utils.viewer.connectedCalendars.invalidate();
+      await utils.viewer.integrations.invalidate();
+    },
+  });
```

Comment:
I am not sure if we should delete credentials when user clicks on Reinstall app because it could be possible that google calendar integration has been expired and user has multiple google calendar accounts connected.

Reply:
and not user doesn't remember which account was expired.

Reply:
Or it could be possible that team's stripe app is invalid but not the personal stripe app.

Reply:
> I am not sure if we should delete credentials when user clicks on Reinstall app because it could be possible that google calendar integration has been expired and user has multiple google calendar accounts connected.

> and not user doesn't remember which account was expired.

Hmm this is tricky. 
We can show the email in the banner eg. "... revoked for Outlook Calendar(some@example.com)", and also show the email in toast message?

Any other suggestions?